File: 1. Two Sum/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let result = [];
    let visited = {};
    
    for (let i = 0; i < nums.length; i++) {
        if ((target - nums[i]) in visited) {
            result = [i, visited[target - nums[i]]];
        }
        visited[nums[i]] = i;
    }
    
    return result;
};

File: arrays/1. Two Sum/twoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        result = []
        # dictionary to store num, index pairs: {nums[i]: i} 
        visited = {}
        
        for i in range (len(nums)):
            if target - nums[i] in visited:
                    result = [i , visited[target - nums[i]]]
            
            visited[nums[i]] = i
        
        return result